test/evp_test.c

```diff
@@ -1074,12 +1086,13 @@ static int cipher_test_run(EVP_TEST *t)
         t->err = "NO_TAG";
         return 0;
     }
+
     for (out_misalign = 0; out_misalign <= 1;) {
         static char aux_err[64];
         t->aux_err = aux_err;
-        for (inp_misalign = (size_t)-1; inp_misalign != 2; inp_misalign++) {
-            if (inp_misalign == (size_t)-1) {
-                /* kludge: inp_misalign == -1 means "exercise in-place" */
+
+        for (inp_misalign = 0; inp_misalign != 2; inp_misalign++) {
```

Comment:
It seems that the way this (existing) loop works is a really horrible pattern.
Couldnt it just be 4 nested for loops? with some simple if's to skip unneeded combinations..

The condition for the frag should just move into a variable to control the for frag loop max.

Reply:
Thanks, fixed